### PR TITLE
tkt-66927: fix(collectd): Do not poll for temperature on drives with SMART disabled

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
@william-gr we'll also need to restart smart daemon in this case, proceed?

**Comment by william-gr** [comment]
@themylogin I think so. But know that you meantion it I am unsure how collectd handles disks going away and showing it? Do you know if it automatically notices it or needs a reload?

**Comment by themylogin** [comment]
@william-gr collectd `disk` plugin calls `geom_stats_snapshot_get()` on each iteration so it does not need a reload.
Our plugin, however, needs to read from database, whether we have or do not have smart enabled for certain drives.

**Comment by themylogin** [comment]
@william-gr ready for review

File: src/middlewared/middlewared/plugins/disk.py [hunk 1]
```diff
@@ -830,6 +831,10 @@ def check_clean(self, disk):
         else:
             disk['disk_identifier'] = await self.middleware.call('datastore.insert', 'storage.disk', disk)
 
+        if await self.middleware.call('service.started', 'collectd'):
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/disk.py]: I am not understanding why we need to check if the service has been started?

Comment (themylogin) [reply src/middlewared/middlewared/plugins/disk.py]: To ensure we won't restart collectd that is not started yet when we are doing boot-time `disk.sync_all`

**Pull request merged**